### PR TITLE
Up-to-date GraphQL plugin packages

### PR DESCRIPTION
**Pull request by iicdii**
### Up-to-date GraphQL plugin packages
- Update `apollo-server-koa` to  **2.16.0**
Some security issue is fixed [Read](https://github.com/apollographql/apollo-server/security/advisories/GHSA-w42g-7vfc-xf37), [Read Full Change log](https://github.com/apollographql/apollo-server/blob/main/CHANGELOG.md)
- Update `graphql` to **^15.3.0**
v14 to v15 Change note is [here](https://github.com/graphql/graphql-js/releases/tag/v15.0.0)
- Update `graphql-playground-middleware-koa` to **^1.6.17**
Docs says all versions before 1.6.15 were vulnerable to user-defined input to koaPlayground(). Read more in the [security notes](https://github.com/prisma/graphql-playground/tree/master/SECURITY.md)
- Update lodash to **4.17.19**
- Update dataloader to **^2.0.0**
- Remove unused pacakge `graphql-tools`

### CODE REVIEW

**Comment by alexandrebodin** [comment]
Hi, there are a lot of major release upgrades. We cannot merge this right away. Can you split this into 2 different PRs ?

**Comment by iicdii** [comment]
@alexandrebodin Okay. I'll make PR for each.

File: packages/strapi-plugin-graphql/package.json [hunk 1]
```diff
@@ -13,14 +13,13 @@
...
     "glob": "^7.1.4",
-    "graphql": "^14.3.0",
+    "graphql": "^15.3.0",
     "graphql-depth-limit": "^1.1.0",
     "graphql-iso-date": "^3.6.1",
-    "graphql-playground-middleware-koa": "^1.6.0",
-    "graphql-tools": "4.0.6",
-    "graphql-type-json": "0.3.1",
+    "graphql-playground-middleware-koa": "^1.6.17",
+    "graphql-type-json": "0.3.2",
```
Comment (alexandrebodin) [inline packages/strapi-plugin-graphql/package.json:22]: Isn't graphql-tools a peer dependencies of one of the other packages ?

Comment (iicdii) [reply packages/strapi-plugin-graphql/package.json:22]: `graphql-tools` is a package that was used to create the GraphQL schema when the GraphQL plugin was first introduced into Strapi.
https://github.com/strapi/strapi/blob/4541b6f30bae74eff240fe8829f54f12d21b6d5c/packages/strapi-plugin-graphql/middlewares/graphql/index.js#L7-L37
The package is now embedded in apollo-server-koa. (https://github.com/apollographql/apollo-server/blob/main/package.json#L108)

The code below shows you calling a function to make a schema from `apollo-server-koa`.
https://github.com/strapi/strapi/blob/35aafde2288e6858a9b1757cf2555a8561421140/packages/strapi-plugin-graphql/services/schema-generator.js#L9

Thus, `graphql-tools` can now be safely removed from `package.json`.

**Comment by alexandrebodin** [comment]
Right now we cannot upgrade dependencies to major without creating breaking changes for our users exploiting features we are not internally. This should go into our next major release.

**Comment by alexandrebodin** [comment]
Hi, Going to close this so we handle upgrades on our side to avoid unwanted breaking changes :) If you want to upgrade minor version this is always welcomed though :)

**Pull request closed without merging**